Add diligent academy sql challenge (#3)
DiligentCorp/rc-tech-assessment: diligent-academy-assessments/backend/database-challenge/README.md

# Database challenge

## Principles and rules

- Please use the prepared `.sql` files to solve the tasks.

## Tasks

> 1. Create the student table with the following fields: `name` and `email`. What columns would you create to represent
>    them? Which types and constraints would make sense for a user table?

Reviewer: Ooops, I made a mistake here. It should refer to the `students` table not the `user` table:
```suggestion
1. Create the student table with the following fields: `name` and `email`. What columns would you create to represent
   them? Which types and constraints would make sense for this table?
```

2. Create the class table, each class has a name, and it have to be **unique**.
   Every student can be in only one class.

>    How would you modify the table if we had a lot of read operations by searching on the class name.

Author: Very true, let me rephrase it.

3. How would you query every student in a given class by name?

4. Create the subject table, each subject has a name. Every student can attend multiple subjects.

>    Where would you store the grades of the students?

Reviewer: Should the subject be part of a class or totally freestanding.

Author: Maybe we are thinking about different things when we talk about classes. I'm thinking about the specific class you are in in a given grade, like how I was once in 9th grade and in class C. I could not find a better word the class.
Subjects are like math and english and stuff.

Reviewer: I see, that's okay.

> 5. How would you get the average grade of a students for a given subject by name?

Reviewer: ```suggestion
5. How would you get the average grade of a students for a given subject by the subject's name?
```
